### DISCUSSION TITLE
Business Process Catalog: unique identification

### DISCUSSION THREAD

**Discussion by pvruyske** [microsoft/dynamics365patternspractices#99] [Q&A] [answered]
We're starting to use the Business Process Catalog througout the lifecycle of Dynamics 365 implementation projects. For example: the BPC is uploaded in a tree structure to Azure DevOps in order to facilitate project scoping, blueprint writing, budgetting, solution delivery follow-up...

However, we currently struggle with quarterly updates to the BPC records. We would like to adjust the associated DevOps records so they are up to date with the most recent BPC and thus Microsoft Learn documentation. It is now very hard to uniquely identify a business process (pattern) in order to update it. Would it be an idea to add a unique identifier througout the Excel so one knows exactly the corresponding record in the previous versions of the BPC? Today it's a manual job to associate both records with each other.

Thanks.
Pieter

**Comment by edupont04** [comment]
@rachel-profitt FYI.

**Comment by pvruyske** [comment]
@rachel-profitt, @edupont04 Have you been able to review my inquiry? :-)

**Reply by rachel-profitt** [reply]
We are excited to announce that the latest version of the catalog which is now available for download includes an ID to sequence the business processes in order.

**Reply by pvruyske** [reply]
Hi @rachel-profitt
That's not exactly what I expected with an ID. I believe an ID should be persistent over time, even if other attributes of the BPC record change. Let's say the sequence of "Define asset major types" changes so it's now positioned before "Define asset types and subtypes". In that scenario the ID of both records should remain the same, so one is able to update it in another database (like Azure DevOps).

![image](https://github.com/microsoft/dynamics365patternspractices/assets/50585884/2d699255-230c-4dbc-8559-b8791d09cb97)

I really like the idea of having a sequence number though! It makes sense to sequence the business process patterns in a logical order (the order in which they are executed in real life).

Thanks.

**Reply by rachel-profitt** [reply]
Thanks for clarifying. There are no current plans to assign our own unique ID to the catalog items as suggested. Although we do have an ID in our database internally, we are not sure that it will add value to customers and partners. We suggest that you upload the catalog to Azure DevOps to get your own unique identifier on each of the items after you have deleted the rows that are not relevant to your project. If we were to add an ID, how do you plan to use this ID?

**Reply by pvruyske** [reply]
As partner, we use the BPC as base structure throughout the complete project lifecycle. Starting with scoping (using Azure DevOps work items), over to blueprint workshops and documentation until key-user training workshops and even after-care.
Within Azure DevOps, we internally maintain a template work item tree structure (with effort estimates,...) combined with a mirrored wiki structure (a page per business process with subtitles per pattern), which is then copied to the customers DevOps project at the start of a new project.
Without a unique ID it is nearly impossible to update this template DevOps tree and documentation structure when BPC records change, especially when they are renamed.
![image](https://github.com/microsoft/dynamics365patternspractices/assets/50585884/3df0df67-afd0-452c-a303-f95a01a6e1dc)

**Comment by rachel-profitt** [comment, marked as answer]
We are planning an Azure DevOps integration in our roadmap that will eliminate this as a problem. Until then, we recommend that you use the Process Sequence ID column as the unique identifier for the rows. This sequence ID should be stable and not changing. We also include a column that indicates if the column is new, updated, or deprecated in the catalog. So that you can consume the updates more easily using a Power Automate Flow for example. Thanks for the feedback about the Wiki, we may consider doing something similar in the future. Are you willing to share what type of content you are delivering in the Wiki by default?

**Reply by pvruyske** [reply]
Looking forward to the ADO integration!
With regards to the wiki contents: we include in the wiki template the basic steps of the business process in business terminology, together with some Dynamics specific information if relevant. During blueprint, the customer processes are described in detail together with the proposed solution and setup in D365FO. During implementation and aftercare, the blueprint wiki is updated when processes or setup change. It acts as an as-built document.
